forch/faucet_event_client.py

```diff
@@ -215,8 +214,16 @@ def next_event(self, blocking=False):
             except Exception as e:
                 LOGGER.info('Error (%s) parsing\n%s*\nwith\n%s*', str(e), line, remainder)
                 continue
-            if self._filter_faucet_event(event):
-                if not self._dispatch_faucet_event(event):
+            event_target = None
+            target_event = None
+            for target in self._handlers:
```

Comment:
Seems that this for loop does not produce any results? Is it an indenting error?

Reply:
It sets target_event and event_target if the target exists in the list of targets we have. They are then used in _filter_faucet_event and _dispatch_faucet_event

Reply:
in general this coding style is problematic b/c it's confusing.  Better to accumulate all results in a list, and then explicitly use [0] from that list. You can/should also use a list comprehension for it, as it's much cleaner overall. 

Also, should minimize the loop (or comprehension) to the minimum necessary, which is selecting the matching "target" value -- that's all that really needs to happen, and then everything else is based on that.